src/docs/development/add-to-app/debugging.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Running, debugging, and hot reload
+short-title: Debugging
+description: How to run, debug, and hot reload your add-to-app Flutter module.
+---
+
+## Running, debugging, and hot reload
+
+Once you've integrated the Flutter module to your project and used Flutter's
+platform APIs to run the Flutter engine and/or UI, you can then build and run
+your Android or iOS app the same way you run normal Android or iOS apps.
+
+However, Flutter is now powering the UI in places where you're showing a
+`FlutterActivity` or `FlutterViewController`.
+
+### Debugging
+
+You may be used to having your suite of favorite Flutter debugging tools
+available to you automatically when running `flutter run` or an equivalent
+command from an IDE. But you can also use all your Flutter [debugging
+functionalities](/docs/testing/debugging) such as hot reload, performance
+overlays, DevTools, and setting breakpoints in add-to-app scenarios.
+
+These functionalities are provided by the `flutter attach` mechanism. `flutter
+attach` can be initiated through different pathways, such as through the SDK's
+CLI tools, through VSCode or IntelliJ/Android Studio.
+
+{% include app-figure.md image="development/add-to-app/debugging/cli-attach.png" caption="flutter attach via terminal" %}
+
+{% include app-figure.md image="development/add-to-app/debugging/vscode-attach.png" caption="flutter attach via VSCode" %}
+
+{% include app-figure.md image="development/add-to-app/debugging/intellij-attach.png" caption="flutter attach via IntelliJ" %}
+
+`flutter attach` can connect as soon as you run your `FlutterEngine`, and will
```

Comment:
nit: create an instance of

Reply:
You meant the engine? I think it only happens after running, not constructing.

Reply:
sg. or "run the engine".

Reply:
I think being specific with the engine is ok here since it's the first time 'engine' is referenced in this page.